### PR TITLE
Feat/authentication/back to sign in

### PR DESCRIPTION
**Pull request by taghizadlaura**
# Sign-Up can go back to Sign-In

## Description
This PR introduces a new composable function for consistent navigation between authentication screens and updates related UI components and tests. It closes issue #300

### UI Component Enhancements:
- Added a new constant `ALREADY_REGISTERED_BUTTON` in `C.kt` to handle the "Already Registered" button.
- Created a new composable function `navigateBetweenAuthScreens` in `AuthenticationComponents.kt` to streamline navigation between authentication screens.

#### Modified
- Replaced the inline navigation row in `SignIn.kt` and added in `SignUp.kt` with the new `navigateBetweenAuthScreens` function to reduce code duplication.

### Testing
- Updated `SignUpScreenTest.kt` to include tests for the new "Already Registered" button and ensure proper navigation.

#### Added
- None

#### Modified
- `SignIn.kt` (replaced inline navigation row with `navigateBetweenAuthScreens` function)
- `SignUp.kt` (added inline navigation row with `navigateBetweenAuthScreens` function)
- `C.kt` (added `ALREADY_REGISTERED_BUTTON` constant)
- `AuthenticationComponents.kt` (added `navigateBetweenAuthScreens` composable function)
- `SignUpScreenTest.kt` (updated tests for the new button and navigation)

#### Removed
- None

## Dependencies Added
- None

## Testing
- Added UI and navigation tests to ensure the "Already Registered" button appears correctly and navigates users to the Sign-In screen as expected.

### CODE REVIEW

File: app/src/main/java/com/android/periodpals/ui/components/AuthenticationComponents.kt [hunk 1]
```diff
@@ -260,3 +264,45 @@ fun AuthenticationSubmitButton(text: String, onClick: () -> Unit, testTag: Strin
...
+ * A composable function that displays a navigation row between authentication screens.
+ *
+ * @param questionText The question text to display.
+ * @param navigateToText The text for the navigation link.
+ * @param screen The screen to navigate to when the link is clicked.
+ * @param testTag The test tag for the navigation link.
+ * @param navigationActions The navigation actions to handle screen navigation.
+ */
+@Composable
+fun navigateBetweenAuthScreens(
```
Comment (Harrish92) [inline app/src/main/java/com/android/periodpals/ui/components/AuthenticationComponents.kt:278]: Composable functions that return Unit should start with an uppercase letter similar to classes. Change the composable function name.

File: app/src/main/java/com/android/periodpals/resources/C.kt [hunk 2]
```diff
@@ -96,6 +96,7 @@ object C {
         const val CONFIRM_PASSWORD_VISIBILITY_BUTTON = "confirmPasswordVisibilityButton"
         const val CONFIRM_PASSWORD_ERROR_TEXT = "confirmPasswordErrorText"
         const val SIGN_UP_BUTTON = "signUpButton"
+        const val ALREADY_REGISTERED_BUTTON = "alreadyRegisteredButton"
```
Comment (Harrish92) [inline app/src/main/java/com/android/periodpals/resources/C.kt:99]: Perhaps rename `ALREADY_REGISTERED_BUTTON` to `ALREADY_REGISTERED_NAV_LINK`.
Do the same for the one in `C.Tag.SignInScreen`

**Review by Harrish92 — APPROVED** [review]
## Summary
Good implementation of a feature that improves the user experience with the app. Just  a few nitpicks.
## Important
#### Code Quality
- The code follows the standard guidelines.
- The composable function `navigateBetweenAuthScreens` should follow the naming convention. See my review comment.
#### Functionality
- The feature works as intended. Good Job.
## Testing
- Nice use of tags, perhaps rename some of them. See my review comment.
## Documentation
- The documentation is well written.
## Suggestions
- See review comments below.
## Steps before PR approved
LGTM

**Review by coaguila — APPROVED** [review]
## Summary
Good job, this should improve how smooth the app feels.
## Important (which includes)
#### Code Quality
Simple and easy to read.
#### Functionality
LGTM
## Testing
## Documentation
Great documentation, very descriptive.
## Suggestions
LGTM

**Pull request merged**